### ISSUE TITLE
unescaped slashes for entry names/paths

### ISSUE THREAD

**Issue by doganoo** [libkeepass/pykeepass#205]
Hey there,

thank you for creating this awesome library!

I have noticed that entry names are not escaped. This leads to wrong path names, as you can see here:

<img width="225" alt="Bildschirmfoto 2020-10-09 um 14 37 06" src="https://user-images.githubusercontent.com/6998611/95583827-02220980-0a3d-11eb-8fd3-c5fa012e4efb.png">

![Bildschirmfoto 2020-10-09 um 14 38 03](https://user-images.githubusercontent.com/6998611/95583948-34336b80-0a3d-11eb-917a-1b21dcbb5daf.png)

Test/Entry is a password in Test/datenbank within the path ../Datenbanken/

Is it possible to escape those names? Or any known workarounds for this?

Thanks in advance!

**EDIT**

```
database = PyKeePass("", "") 

for group in database.groups:
    print(group.name)
    print("----------------")
    for entry in group.entries:
        print(entry.path)
```

**Comment by pschmitt** [comment]
How do you create these entries?

**Comment by doganoo** [comment]
> How do you create these entries?

using KeepassXC "new group"

**Comment by Evidlo** [comment]
It's more an issue about how we are representing paths.  The KeePass spec allows `/` inside of entry titles and group names, and since we use it as a delimiter, things can be a bit confusing.

**Comment by KeyWeeUsr** [comment]
But then it's as good delimiter as any other due to keepass allowing almost everything in there. We can do spaces between the paths or choose a new delimiter (e.g. `>` )and the same issue will be opened later.

We might also do some bracketing for visibility e.g.:

    [Root]/[Group/with/slash]/[Title]

**Comment by Evidlo** [comment]
The safest thing would be to return a list instead of a string, but this is
a breaking change.  What are you using paths for?

**Comment by KeyWeeUsr** [comment]
@Evidlo Not necessarily, we can just add a new attribute and keep the original path in place. This way it won't be a breaking change and people who'd want to use the list would be able to.

**Comment by doganoo** [comment]
@Evidlo whats about represeting as a graph?

**Comment by KeyWeeUsr** [comment]
@doganoo Do you mean as a CLI tool like in #207 ?

**Comment by doganoo** [comment]
> @doganoo Do you mean as a CLI tool like in #207 ?

@KeyWeeUsr Yes exactly :-)

**Comment by Evidlo** [comment]
I think I'll make this a breaking change and release it with the next major version.

**Comment by doganoo** [comment]
cool, do you have a timeline for that?

**Comment by Evidlo** [comment]
Check out https://github.com/libkeepass/pykeepass/tree/pathlist and let me know what you think.

I may do a major release in the next couple of weeks.

**Comment by doganoo** [comment]
@Evidlo any updates here? :)

**Comment by Evidlo** [comment]
`pathlist` is now merged into master.  Just waiting on https://github.com/libkeepass/pykeepass/pull/204 before making a release.

**Comment by spaetz** [comment]
As a downstream user I have to say that I would have preferred a fixed up path2 attribute instead of a breaking change in order to avoid version dependent-behavior changes. Especially, given that the version variable itself is not guaranteed to be there (as it was just introduced a version earlier).
That having said, we can certainly cope with the change...
